### ISSUE TITLE
Mailing list messages should auto-generate Notes and replies should be saved as comments

### ISSUE THREAD

**Issue by jywarren** [jywarren/plots#12]
http://drupal.org/project/listhandler

Copying in info from duplicate issues:

Point: Sara
Team: RJ, Stewart, Shannon, Jeff
Importance: Medium- October suggested

Including: 
- How to integrate mailing list with homepage - takes up big bandwidth on the homepage
- clearer ways that the mailing list and website interface
- possibly feeds that mention a place or tool are added into online “archive” of pages

**Goals:** The goal of this project is to increase interaction between the mailing list and the website by facilitating the push and pull of data and conversations between these two communication methods. Specifically we will review and revise current cross communication such as the mailing list feed on the website.

**Comment by jywarren** [comment]
More notes:

> I think this might be tough, but I have wanted to do it for a long time. +1 but no clear idea of how to get it done... :-( 
> Well, it might involve moving off of Google Groups and using a Drupal module called Organic Groups:
> http://drupal.org/project/og_mailinglist
> Or creating a "forums" area and installing these: http://drupal.org/node/40777
> RJ will know more. He had some ideas posted for programming bottleneck, among them attending Drupal events and trying to attract help there. Anyhow if you can, please post the above links/notes to the github issue on mailing list integration, which already exists...

**Comment by rjcorwin** [comment]
I wrote about this somewhere but I'm not sure where that was now... Oh well.  From what I remember....

We'll need some way to add a tag to conversation that allows code to know which tool that conversation should show up in.
1. Use a mailing list for each tool.  The mailing list itself is the tag.  We could go with Drupal's organic groups as mailing lists or create more google groups.  The former, while potentially being a more unified/user-friendly option, is more expensive given we would have to maintain it ourselves.
2. Use one mailing list and create an editorial workflow that allows a user(s) to easily tag conversations.
3. Use one mailing list and use some AI to figure out what tags should be added to each conversation.

There is another side to this, users and the credit that users get for posting.  If everyone is using the same email address on the list as they are on the Drupal PLOTS site, then we can match them, but I can't imagine that would be the case all the time.  However, perhaps it's just research notes we care about giving (quantifiable) credit to and we can pass on the conversation.

**Comment by jywarren** [comment]
I think we could simplify this by:
- starting with the problem of integrating mailing lists into the site: potentially each threaded conversation could be a node or forum topic, with all followups recorded as comments? I'm not sure how listhandler works but would like to learn more
- just using tags to manually tag any conversation (community tags makes it easy for anyone to participate in this tagging already) which would then make the correct nodes (threaded conversations) show up on the tool/place pages if we want

I think the first of these is much harder but also more important.

**Comment by jywarren** [comment]
I remember the clincher being the part about followup emails becoming comments. I wonder if we could use http://drupal.org/project/mailcomment or some variation on it?

I think mailcomment is really for responses to _notifications_ becoming comments, not responses to emails. Potentially Listhandler would just create a new node for each response, which might be a mess. Some hybridization might be able to detect the same subject line (disregarding prefixes like Re: or Fwd: or Re: Re: ) and create a comment instead of a new note. Worth googling for someone who's done this...

**Comment by rjcorwin** [comment]
(looks like Jeff posted while I was writing, posting anyways :) )

In the past I've used Mailhandler module (http://drupal.org/project/mailhandler) for posting nodes via email and Mail Comment module (http://drupal.org/project/mailcomment) for replying to posts via email.  I've used those modules a lot when setting up Open Atrium installations for teams.  The how-to here (https://community.openatrium.com/documentation-en/node/2286)  gives you a pretty good idea of how to set something like this up but... I'm not so sure it will work with a mailing list as it is intended to work with a mailbox... I know I'm not thinking this all the way through.

I haven't had a chance to look too deeply into OG Mailinglist (http://drupal.org/project/og_mailinglist) but it definitely looks like it's worth looking into.

**Comment by jywarren** [comment]
OG Mailinglist seems to be very close to what we want... in that replies become comments. I wonder if we tried setting it up and subscribing something like "talk@publiclaboratory.org" or "list@publiclaboratory.org" (as the OGM address) to the Google Group, we'd be able to see this working on a test site, to assess if it's a workable option. I almost think that would work out of the box...

**Comment by rjcorwin** [comment]
Ok, now I remember what I was thinking.  Using Mailhandler+Mailcomment might work if we create a dummy account (googlegroup@publiclaboratory.org), subscribe it to the public labs list on google, and then point mailhandler towards the inbox of googlegroup@publiclaboratory.org.

**Comment by jywarren** [comment]
we keep crossing in cyberspace. I was thinking the same thing for OGM, but if you have more experience with Mailhandler + Mailcomment, let's give it a try... I'd like to try on a separate system so we don't spam the list or mess up the website with such a big additional feature.

**Comment by rjcorwin** [comment]
Agreed.  Sandboxing this first is a good idea.  I won't have time to try this until late next week though.  Cleanweb Hackathon is this weekend so I'll be busy building online AI for greenhouses.

**Comment by jywarren** [comment]
no problem. if I have a chance i'll try setting up a test server; now I've
done it once for privatelaboratory.org, it won't be as big a deal.

On Fri, May 4, 2012 at 3:48 PM, R.J. Steinert <
reply@reply.github.com
